### PR TITLE
BUG: Mixed DataFrame with Extension Array incorrect aggregation

### LINKED ISSUE #34520: BUG: Mixed DataFrame with Extension Array incorrect aggregation
Surprised to not get an aggregation for "b" with the Int64 dtype

```python
>>> df = pd.DataFrame([["a", 1]], columns=list("ab"))
>>> df.sum()
a    a
b    1
dtype: object
>>> df.astype({"b": "Int64"}).sum()
a    a
dtype: object
```

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] closes #34520 by reverting #32950

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
cc @jbrockmendel

File: pandas/core/frame.py [hunk 1]
```diff
@@ -8539,51 +8539,42 @@ def blk_func(values):
...
             data = self
             values = data.values
-
             try:
                 result = f(values)
 
             except TypeError:
                 # e.g. in nanops trying to convert strs to float
 
+                # try by-column first
```
Comment (jbrockmendel) [inline pandas/core/frame.py:8551]: why does this need to be moved?  if it is moved, then the "try by-column first" comment is no longer accurate

Comment (simonjayhawkins) [reply pandas/core/frame.py:8551]: > the "try by-column first" comment is no longer accurate

This comment was there originally before the move in https://github.com/pandas-dev/pandas/pull/32950/files.

> why does this need to be moved?

This PR reverts a change that caused a regression. The PR that caused the regression is labelled as a clean. This PR is in response to https://github.com/pandas-dev/pandas/issues/34730#issuecomment-653208905.

**Review by jreback — APPROVED** [review]
ok, here, @jbrockmendel ok with this? (certainly can follow up with a cleanup PR)

**Comment by jreback** [comment]
ping @jbrockmendel

File: pandas/tests/arrays/integer/test_function.py [hunk 2]
```diff
@@ -133,6 +133,15 @@ def test_integer_array_numpy_sum(values, expected):
     assert result == expected
 
 
+def test_mixed_frame_with_integer_sum():
+    # https://github.com/pandas-dev/pandas/issues/34520
+    df = pd.DataFrame([["a", 1]], columns=list("ab"))
+    df.astype({"b": "Int64"})
```
Comment (jbrockmendel) [inline pandas/tests/arrays/integer/test_function.py:139]: should this be `df = df.astype`?

**Comment by jbrockmendel** [comment]
I think there are better fixes available.  The underlying problem is that we're calling `nanops.nansum` on an IntegerArray and that raises `ValueError`.  So we should either make `nanops.nansum` handle `IntegerArray` or change the definition of `f` to dispatch to `EA._reduce` cc @jorisvandenbossche has been working on the latter approach

File: pandas/tests/arrays/integer/test_function.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/tests/arrays/integer/test_function.py]: fixed

**Comment by simonjayhawkins** [comment]
closing in favour of #32867

**Pull request closed without merging**